Source/MIKMIDIObject.m

```diff
@@ -132,6 +140,12 @@ - (MIDIUniqueID)uniqueID
 	return _uniqueID;
 }
 
+- (void)setUniqueID:(MIDIUniqueID)uniqueID
+{
+	_uniqueID = uniqueID;
+	_hash = @(uniqueID).hash;
```

Comment:
Did profiling indicate that caching the hash value actually makes an important performance difference in a real application? If not, I'd probably prefer the simpler approach of just doing `-(NSUInteger)hash { return [@(self.uniqueID) hash]; }`